core/src/test/java/io/parsingdata/metal/TreeTest.java

```diff
@@ -57,10 +56,16 @@
         seq("tree",
             def("head", con(1), eq(con(HEAD))),
             def("nr", con(1)),
-            def("left", con(1)),
-            pre(sub(token("tree"), last(ref("left"))), not(eq(last(ref("left")), con(0)))),
-            def("right", con(1)),
-            pre(sub(token("tree"), last(ref("right"))), not(eq(last(ref("right")), con(0))))
+            cho(def("left_terminator", con(1), eq(con(0))),
```

Comment:
The reason I'm asking is that in practice I would be more likely to write the original style (`def+pre(sub)`), that is: "read pointer, sub elsewhere based on the pointer's value".